### PR TITLE
Add rope.py fixed issue #3456

### PR DESCRIPTION
**Pull request by ulshell**
**Fixes issue:**

**Changes:**

### CODE REVIEW

File: code/data_structures/src/tree/binary_tree/rope/rope.py [hunk 1]
```diff
@@ -57,7 +57,9 @@ def __init__(self, text=None):
         else:
             self.head = node()
 
-    def Index(self, ind, node=self.head):
+    def Index(self, ind, node=None):
+        if node == None:
```
Comment (arnavb) [inline code/data_structures/src/tree/binary_tree/rope/rope.py]: Instead check if `is None`, not `== None`.

Comment (ulshell) [reply code/data_structures/src/tree/binary_tree/rope/rope.py]: yes I have changed it.

**Comment by ulshell** [comment]
@arnavb kindly merge my code of issue #3456

**Comment by arnavb** [comment]
@ulshell I cannot merge, unfortunately; Wait for @AdiChat!

**Comment by ulshell** [comment]
@arnavb thanks

**Comment by ulshell** [comment]
@arnavb my code in not merging please contact @AdiChat

**Comment by arnavb** [comment]
@ulshell Please have some patience! Aditya has to go through a lot of PRs every day, and he may not have enough time every day to go through _every_ PR. It's hard when you get around 300-400 notifications in a day! 😓 Just wait a few more days, and let Aditya get to your PR!

**Comment by AdiChat** [comment]
Looks great 😍 

There has been some delay but we hope you are enjoying your journey 🚆  with **OpenGenus**

positive_opengenus_first

**Pull request merged**